tests/test_maintel_take_image_anycam.py

```diff
@@ -0,0 +1,403 @@
+):
+    async def basic_make_script(self, index):
+        self.script = TakeImageAnyCam(index=index)
+
+        # Mock MTCS instances and its methods
+        self.script.mtcs = unittest.mock.AsyncMock()
+        self.script.mtcs.start_task = utils.make_done_future()
+
+        self.script.lsstcam = unittest.mock.AsyncMock()
+        self.script.lsstcam.start_task = utils.make_done_future()
+
+        self.script.comcam = unittest.mock.AsyncMock()
+        self.script.comcam.start_task = utils.make_done_future()
+
+        self.script.generic_cams = []
+        # Mock specific behaviors for each Generic Camera
+        for i in [101, 102, 103]:  # Adjust indexes as needed
+            gencam = unittest.mock.MagicMock(name=f"GenericCam_{i}")
+            gencam.start_task = utils.make_done_future()
+            gencam.name = f"GenericCam_{i}"
+            self.script.generic_cams.append(gencam)
+
+        return (self.script,)
+
+    @contextlib.asynccontextmanager
+    async def make_dry_script(self):
+        """Context manager for setting up the script for dry run testing."""
+        async with self.make_script():
+            # Use mock_cameras to setup all camera
+            self.mock_cameras()
+
+            # Mock the take_images_with_camera method
+            self.script.take_images_with_camera = unittest.mock.AsyncMock()
+
+            yield
+
+    async def mock_cameras(self):
+        # Mock MTCS with immediate start_task completion
+        self.script.mtcs = unittest.mock.AsyncMock()
+        self.script.mtcs.start_task = utils.make_done_future()
+        self.script.mtcs.assert_liveliness = unittest.mock.AsyncMock()
+
+        # Mock specific behaviors for ComCam and LSSTCam
+        for cam_name in ["ComCam", "LSSTCam"]:
+            camera = unittest.mock.MagicMock(name=f"{cam_name}")
+            camera.start_task = utils.make_done_future()
+            camera.assert_liveliness = unittest.mock.AsyncMock()
+            camera.assert_all_enabled = unittest.mock.AsyncMock()
+            camera.take_imgtype = unittest.mock.AsyncMock()
+            camera.setup_instrument = unittest.mock.AsyncMock()
+            camera.read_out_time = 2
+            camera.shutter_time = 0.1
+            setattr(self.script, cam_name.lower(), camera)
+
+        # Mock generic cameras
+        self.script.generic_cams = []
+        for i in [101, 102, 103]:
+            gencam = unittest.mock.MagicMock(name=f"GenericCam_{i}")
+            gencam.start_task = utils.make_done_future()
+            gencam.assert_liveliness = unittest.mock.AsyncMock()
+            gencam.assert_all_enabled = unittest.mock.AsyncMock()
+            gencam.take_imgtype = unittest.mock.AsyncMock()
+            gencam.read_out_time = 1
+            gencam.shutter_time = 0.1
+            self.script.generic_cams.append(gencam)
+
+            yield
+
+    async def configure_and_mock_camera_setups(self, **config):
```

Comment:
I don't think you need this. You are already creating the mocks in the `basic_make_script` so just call `configure_script`